# YAML Document starts with ---
# Comments start with #
---
  # Training
  training_steps: 200000      # The number of training steps to perform

  # Save params
  save_freq: 1000
  save_prefix: ddpg_model_nov2_doc
  trained_model_name: trained_model_nov2_doc
  save_replay_buffer: True

  # Load model params
  load_model: False
  model_name: ddpg_model_15000_steps

  # Logging parameters
  log_folder: DDPG_model_nov2_doc
  log_interval: 4 # The number of episodes between logs
  reset_num_timesteps: False # If true, will reset the number of timesteps to 0 every training

  # Use custom policy
  use_custom_policy: False
  policy_params:
    net_arch: [400, 300] # List of hidden layer sizes
    activation_fn: relu  # relu, tanh, elu or selu
    features_extractor_class: FlattenExtractor # FlattenExtractor, BaseFeaturesExtractor or CombinedExtractor
    optimizer_class: Adam # Adam, Adadelta, Adagrad, RMSprop or SGD

  # Use action_noise
  use_action_noise: True # For now only Gaussian noise is supported
  action_noise:
    mean: 0.0
    sigma: 0.01

  # HER parameters
  use_HER: True
  her_params:
    n_sampled_goal: 4
    goal_selection_strategy: future # future, final or episode
    online_sampling: True
    max_episode_length: 100

  # DDPG parameters
  ddpg_params:
    learning_rate: 0.001
    buffer_size: 1000000
    learning_starts: 100
    batch_size: 100
    tau: 0.005
    gamma: 0.99
    gradient_steps: -1
    train_freq:
      freq: 20
      unit: step # episode or step
    seed: 0
